test/RulesTest.java

```diff
@@ -0,0 +1,45 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class RulesTest {
+
+    private RockRule rockRule;
+
+    @Before
+    public void setUp() throws Exception {
```

Comment:
Please change your intellij template to remove the `throws Exception`. Checked exceptions are the worst, and we should avoid them as much as possible which means methods that advertise that they throw checked exceptions should be avoided.